### ISSUE TITLE
Gaps in Graphs

### ISSUE THREAD

**Issue by gb5256** [netdata/netdata#9064]
Hello,
I expierence gaps in the graphs:

<img width="1121" alt="image" src="https://user-images.githubusercontent.com/324658/82076900-c6c61d80-96de-11ea-9ada-b2987cc0f6b6.png">

Looking at the netdata error logs I see lots of missing heartbeats:
2020-05-15 04:30:04: netdata ERROR : STATSD : heartbeat missed 1668006 monotonic microseconds
2020-05-15 04:30:04: netdata ERROR : PLUGIN[cgroups] : heartbeat missed 1668006 monotonic microseconds
2020-05-15 04:30:04: netdata ERROR : PLUGIN[diskspace] : heartbeat missed 1668006 monotonic microseconds
2020-05-15 04:30:04: netdata ERROR : PLUGIN[proc] : heartbeat missed 1993007 monotonic microseconds
2020-05-15 04:30:06: netdata ERROR : PLUGIN[cgroups] : heartbeat missed 1050004 monotonic microseconds
2020-05-15 04:30:06: netdata ERROR : PLUGIN[diskspace] : heartbeat missed 1050004 monotonic microseconds
2020-05-15 04:30:06: netdata ERROR : STATSD : heartbeat missed 889004 monotonic microseconds

I can not force to reproduce the error. But what I can see is that if it happens, my whole server seems to freeze for some seconds.
As netdata shows no data during these moments, it is hard for me to see what is happening or starting this...

Any ideas?
I am running latest nightly on Ubuntu 18.04

Thanks for any comments!
gb5256

**Comment by thiagoftsm** [comment]
Hi @gb5256 ,

Do you have slave connections on this server? Or is it a standalone Netdata ? 

Another doubt, please, can you share the values that you are using for `memory mode`, `page cache size` and `dbengine disk space`?

best regards!

**Comment by gb5256** [comment]
Hi, thanks for your help..
this is a standalone Netdata server.
The values for memory mode etc are untouched, so they are the netdata default values:
        # memory mode = dbengine
        # page cache size = 32
        # dbengine disk space = 256

**Comment by thiagoftsm** [comment]
Hi @gb5256 ,

Considering that default values are probably creating problems for you, it is possible you need to adjust your environment, please, take a look on our [documentation](https://learn.netdata.cloud/docs/agent/database/engine/#file-descriptor-requirements).

Best regards!

**Comment by gb5256** [comment]
Hi @thiagoftsm,
thanks for you remarks. I have played around with several settings, but not of them made any difference. I have then found somebody complaining about ubuntu 18.4 freezing problems with kernel 4.x So I have upgraded to kernel 5.x as that comes with some hardware improvements. That has actually solved the issues. So it looks like the gaps where not related to netdata, but just to the whole machine which got into a freeze and that way also netdata froze and of course could not get any data to display. Now all looks fine. THANKS !
